dotnet/src/Extensions/Extensions.UnitTests/Planning/ActionPlanner/ActionPlannerTests.cs

```diff
@@ -149,4 +179,41 @@ private static Mock<ISKFunction> CreateMockFunction(FunctionView functionView)
         mockFunction.Setup(x => x.SkillName).Returns(functionView.SkillName);
         return mockFunction;
     }
+
+    private Mock<ISkillCollection> CreateMockSkillCollection()
+    {
+        var functions = new List<(string name, string skillName, string description, bool isSemantic)>()
+        {
+            ("SendEmail", "email", "Send an e-mail", false),
+            ("PullsList", "GitHubSkill", "List pull requests", true),
+            ("RepoList", "GitHubSkill", "List repositories", true),
+        };
+
+        var functionsView = new FunctionsView();
+        var skills = new Mock<ISkillCollection>();
+        foreach (var (name, skillName, description, isSemantic) in functions)
+        {
+            var functionView = new FunctionView(name, skillName, description, new List<ParameterView>(), isSemantic, true);
+            var mockFunction = CreateMockFunction(functionView);
+            functionsView.AddFunction(functionView);
+
+            mockFunction.Setup(x =>
+                    x.InvokeAsync(It.IsAny<SKContext>(), It.IsAny<CompleteRequestSettings>(), It.IsAny<CancellationToken>()))
+                .Returns<SKContext, CompleteRequestSettings, CancellationToken>((context, settings, CancellationToken) =>
+                {
+                    context.Variables.Update("MOCK FUNCTION CALLED");
+                    return Task.FromResult(context);
+                });
+
+            skills.Setup(x => x.GetFunction(It.Is<string>(s => s == skillName), It.Is<string>(s => s == name)))
+                .Returns(mockFunction.Object);
+            ISKFunction? outFunc = mockFunction.Object;
+            skills.Setup(x => x.TryGetFunction(It.Is<string>(s => s == skillName), It.Is<string>(s => s == name), out outFunc)).Returns(true);
+        }
+
+        skills.Setup(x => x.GetFunctionsView(It.IsAny<bool>(), It.IsAny<bool>())).Returns(functionsView);
+        return skills;
+    }
+
+    private readonly string _validPlanString = "Here is a possible plan to accomplish the user intent:\n\n{\"plan\":{\n\"rationale\": \"the list contains a function that allows to list pull requests\",\n\"function\": \"GitHubSkill.PullsList\",\n\"parameters\": {\n\"owner\": \"microsoft\",\n\"repo\": \"semantic-kernel\",\n\"state\": \"open\"\n}}}\n\nThis plan uses the `GitHubSkill.PullsList` function to list the open pull requests for the `semantic-kernel` repository owned by `microsoft`. The `state` parameter is set to `\"open\"` to filter the results to only show open pull requests.";
```

Comment:
This can be constant with name `ValidPlanString`. Since we declared separate field for this value, we can also reuse it in `ExtractsAndDeserializesWellFormedJsonFromPlannerResult` test, it looks like the values are the same. 

Improvement ideas: in IDE it's hard to read this value, you have to use horizontal scroll :) We can move this string to separate .txt file and read from it, or break string in multiple lines for better readability. For other long strings in this file (located in tests) we can use line break.

Reply:
Made this a `const` and added actual newlines to the invalid plan string in one of the tests.